### DISCUSSION TITLE
Sentinel-1 orbit file change starting 1/31

### DISCUSSION THREAD

**Discussion by Xiaohua-Eric-Xu** [gmtsar/gmtsar#629] [Announcements]
Extra empty lines introduced to .EOF files starting 1/31, so newer data may fail with older code (reported by Wanpeng). Please update your master branch or 6.2 to the newer version where a temporary fix is made on xml.c to account for this change.

**Comment by Xiaohua-Eric-Xu** [comment]
Make sure you `make clean` before compiling the new code, as the update is made on xml.c which is built as a library.

**Reply by xt-xc-123** [reply]
Hello,I want to know how to update gmtsar ,is there any scripts to do it,thank you.

**Reply by Xiaohua-Eric-Xu** [reply]
If you installed from github, simply go to your GMTSAR directory and run `git pull`, then you can update your code.

**Reply by xt-xc-123** [reply]
> If you installed from github, simply go to your GMTSAR directory and run `git pull`, then you can update your code.

Thank you,I have completed it successfully.

**Comment by krkrueger** [comment]
I did a git pull, and make clean in my directory.  But I am still having some issues with S1 orbit files in 2023.

I am getting the following when running create_frame_tops:
Working on subswath 1 ...
Reading in 6036 lines of info from XML...
Writing 17 lines for orbit...
Not an well formatted XML file...-1, -1
Unable to find designated string...after 0 searches..
NO orbit coverage in the selected file...

I have a handful of images dating back to 2021, they all work until I get to one that is 01/21/2023.

Anything I should check to make sure that I am using the right updated code? Thanks!

**Reply by krkrueger** [reply]
just manually deleting the extra spaces in the orbit files seems to work, but obviously not the ideal situation!

**Reply by Xiaohua-Eric-Xu** [reply]
Hmm, did you ran `make` and `make install` afterwards? Also, make sure you are using 6.2 or master. You could check the version with `git status`.

**Reply by krkrueger** [reply]
Yeah the make/make install have fatal errors regarding the tiffio.h file not being present.

When I check gmt it says I am using version 6.2. When I do git status, I see:

# On branch 6.1
# Changes not staged for commit:
#   (use "git add <file>..." to update what will be committed)
#   (use "git checkout -- <file>..." to discard changes in working directory)
#
#	modified:   preproc/ERS_preproc/ers_line_fixer/ers_line_fixer.c
#
# Untracked files:
#   (use "git add <file>..." to include in what will be committed)
#
#	preproc/ALOS_preproc/ALOS_baseline/ALOS_baseline
no changes added to commit (use "git add" and/or "git commit -a")

Which might be the issue there, I see 6.1.

**Reply by krkrueger** [reply]
and unfortunately it doesn't look like just deleting the spaces fixed the issue, the next step running preprocessing I get a bunch of nan! in the log output, and only about half the images in in the baseline_table.dat have data in the last 2 columns. (not just in 2023)

**Reply by Xiaohua-Eric-Xu** [reply]
You  probably need to switch to a newer branch of GMTSAR, e.g. 6.2 or master. The update were not made on any earlier versions. Try `git switch 6.2` to check out the newer branch and then compile the code.

**Reply by krkrueger** [reply]
This was the fix that was required.  But just in case someone else pops in, git switch 6.2 didn't work for me, I had to use git checkout 6.2.  Everything (at the moment!) seems to be working as intended!

**Comment by krkrueger** [comment]
Baseline map looks like the following.
![baselineEX](https://github.com/gmtsar/gmtsar/assets/41383336/634b70a5-0018-4d06-82aa-7f37e8a5c883)

**Comment by shubhayanroychowdhury** [comment]
@Xiaohua-Eric-Xu 
Hello, I have faced the same problem. 


Successfully opened junk1 
Reading in 5568 lines of info from XML...
Writing 16 lines for orbit...
Writing 53 lines of precise orbit for the LED file...
Successfully opened junk1 
Reading in 5568 lines of info from XML...
Writing 16 lines for orbit...
Not an well formatted XML file...-1, -1
Unable to find designated string...after 0 searches..
NO orbit coverage in the selected file...
Successfully opened junk1 
interpolation point outside of data constraints, 996455.582522 0.000000 0.000000
cat: junk2: No such file or directory
rm: cannot remove 'junk2': No such file or directory
Reading in 5582 lines of info from XML...
Writing 16 lines for orbit...
Not an well formatted XML file...-1, -1
Unable to find designated string...after 0 searches..
NO orbit coverage in the selected file...
Successfully opened junk1 
interpolation point outside of data constraints, 2033255.731872 0.000000 0.000000
cat: junk2: No such file or directory
rm: cannot remove 'junk2': No such file or directory
Reading in 5568 lines of info from XML...
Writing 16 lines for orbit...
Not an well formatted XML file...-1, -1
Unable to find designated string...after 0 searches..
NO orbit coverage in the selected file...
Successfully opened junk1 
interpolation point outside of data constraints, 3070055.180459 0.000000 0.000000
cat: junk2: No such file or directory
rm: cannot remove 'junk2': No such file or directory
Reading in 5582 lines of info from XML...
Writing 16 lines for orbit...
Not an well formatted XML file...-1, -1
Unable to find designated string...after 0 searches..
NO orbit coverage in the selected file...
Successfully opened junk1 
interpolation point outside of data constraints, 4106854.703190 0.000000 0.000000
cat: junk2: No such file or directory
rm: cannot remove 'junk2': No such file or directory
Reading in 5582 lines of info from XML...
Writing 16 lines for orbit...
Not an well formatted XML file...-1, -1
Unable to find designated string...after 0 searches..
NO orbit coverage in the selected file...
Successfully opened junk1 
interpolation point outside of data constraints, 5143654.926941 0.000000 0.000000
cat: junk2: No such file or directory
rm: cannot remove 'junk2': No such file or directory
Reading in 5582 lines of info from XML...
Writing 16 lines for orbit...
Not an well formatted XML file...-1, -1
Unable to find designated string...after 0 searches..
NO orbit coverage in the selected file...
Successfully opened junk1 
interpolation point outside of data constraints, 6180454.720128 0.000000 0.000000
cat: junk2: No such file or directory
rm: cannot remove 'junk2': No such file or directory
Reading in 5568 lines of info from XML...
Writing 16 lines for orbit...
Not an well formatted XML file...-1, -1
Unable to find designated string...after 0 searches..
NO orbit coverage in the selected file...
Successfully opened junk1 
interpolation point outside of data constraints, 7217255.120107 0.000000 0.000000
cat: junk2: No such file or directory
rm: cannot remove 'junk2': No such file or directory
Reading in 5568 lines of info from XML...
Writing 16 lines for orbit...
Not an well formatted XML file...-1, -1
Unable to find designated string...after 0 searches..
NO orbit coverage in the selected file...
Successfully opened junk1 
interpolation point outside of data constraints, 8254055.123557 0.000000 0.000000
cat: junk2: No such file or directory
rm: cannot remove 'junk2': No such file or directory
Reading in 5568 lines of info from XML...
Writing 16 lines for orbit...
Not an well formatted XML file...-1, -1
Unable to find designated string...after 0 searches..
NO orbit coverage in the selected file...
Successfully opened junk1 
interpolation point outside of data constraints, 9290855.405574 0.000000 0.000000
cat: junk2: No such file or directory
rm: cannot remove 'junk2': No such file or directory
Reading in 5568 lines of info from XML...
Writing 16 lines for orbit...
Not an well formatted XML file...-1, -1
Unable to find designated string...after 0 searches..
NO orbit coverage in the selected file...


After updating GMTSAR from 6.0 to 6.2, the problems remain same. I want to process the sentinel data from 1st January 2020 to 30 April 2023. I am able to process this step upto 2022. Please help me how to process the data of 2023. Thank you for your time and consideration.

**Reply by Xiaohua-Eric-Xu** [reply]
This is likely from the same issue as in this discussion. https://github.com/gmtsar/gmtsar/discussions/629

You'll need to 1) switch to master or 6.2, and run a `git pull`. 2) `make clean`. 3) `make` and `make install`.

**Reply by shubhayanroychowdhury** [reply]
Thank you so much. I will let you know.

**Reply by shubhayanroychowdhury** [reply]
@Xiaohua-Eric-Xu 
Thank you so much. Now i am able to process the 2023 data.
But I got another error now.

![Screenshot from 2023-07-03 18-38-40](https://github.com/gmtsar/gmtsar/assets/129718401/3456b8b6-944c-4f4c-a92e-aec64445b3a8)
![Screenshot from 2023-07-03 18-39-11](https://github.com/gmtsar/gmtsar/assets/129718401/25fc2fe7-488b-442e-ba42-fd1a840030a5)
In my data.in file, there are 10 files. After run this command **preproc_batch_tops.csh data.in dem.grd 2**, when I open baseline_table.date, it contains a total of 20 files. Why does this happen?. It should be a equal number. right? 
Also, when I opened the raw folder, total 10 SLC files were generated.

**Reply by Xiaohua-Eric-Xu** [reply]
I am not exactly sure what happened, but you could run `get_baseline_table.csh` to create a baseline_table.dat.

**Comment by shubhayanroychowdhury** [comment]
@Xiaohua-Eric-Xu 
![Screenshot from 2023-07-07 22-20-47](https://github.com/gmtsar/gmtsar/assets/129718401/a73bdaa5-da0b-417e-b2af-9db0f5e48501)
When I run get_baseline_table.csh, I got this.

**Reply by Xiaohua-Eric-Xu** [reply]
You may need to add the input prmlist (with `ls *ALL*PRM > prmlist` in your raw directory) and the name of your master PRM file.

**Comment by shubhayanroychowdhury** [comment]
@Xiaohua-Eric-Xu 
Thank you so much for your guidance. Now I am able to process the data from 2020 to 2023. 
Thank you.

**Comment by WY-SVG** [comment]
Hello, I'm having a problem running preproc_batch_tops.csh data.in dem.grd 1. Please help me.
[Uploading DESKTOP.docx…]()

**Reply by Xiaohua-Eric-Xu** [reply]
@WY-SVG , I think you need to upload a log file. The link you attached only redirects to this issue webpage.
